lib/mongo/protocol/serializers.rb

```diff
@@ -50,7 +50,7 @@ module Header
         # @param value [ String ] Header value to be serialized.
         #
         # @return [ String ] Buffer with serialized value.
-        def self.serialize(buffer, value)
+        def self.serialize(buffer, value, validating_keys = nil)
```

Comment:
Yes I would reaise the BSON dependency to 4.1 which would force this change to be in driver 2.3.0